File: 0148-sort-list/0148-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode sortList(ListNode head) {
        ArrayList<Integer> ans = new ArrayList<>();
        ListNode temp = head;
        while(temp!=null){
            ans.add(temp.val);
            temp=temp.next;
        }

        Collections.sort(ans);
        ListNode dummy = new ListNode(0);
        ListNode dummy1 = dummy;
        for(int v : ans){
            dummy1.next = new ListNode(v);
            dummy1 = dummy1.next;
        }
        return dummy.next;
        
    }
}

File: 0148-sort-list/0148-sort-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def sortList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        list = []
        temp = head
        while temp != None:
            list.append(temp.val)
            temp = temp.next
        list.sort()
        dummy = ListNode(0)
        dummy1 = dummy
        for num in list:
            dummy1.next = ListNode(num)
            dummy1 = dummy1.next
        return dummy.next
        